src/ICS26Router.sol

```diff
@@ -147,18 +147,29 @@ contract ICS26Router is IICS26Router, IICS26RouterErrors, Ownable, ReentrancyGua
         }
 
         bytes[] memory acks = new bytes[](1);
-        acks[0] = getIBCApp(payload.destPort).onRecvPacket(
+        bool recvSuccess = true;
+        try getIBCApp(payload.destPort).onRecvPacket(
             IIBCAppCallbacks.OnRecvPacketCallback({
                 sourceChannel: msg_.packet.sourceChannel,
                 destinationChannel: msg_.packet.destChannel,
                 sequence: msg_.packet.sequence,
                 payload: payload,
                 relayer: _msgSender()
             })
-        );
-        require(acks[0].length != 0, IBCAsyncAcknowledgementNotSupported());
+        ) returns (bytes memory ack) {
+            require(ack.length != 0, IBCAsyncAcknowledgementNotSupported());
+
+            acks[0] = abi.encodePacked(
+                ack
+            );
+        } catch (bytes memory errorData) {
+            recvSuccess = false;
+            acks[0] = abi.encodePacked(
+                errorData
+            );
```

Comment:
I don't get this. We are returning an error ack if callback fails, couldn't this happen due to gas reasons?

Reply:
AHh yes, i forgot about gas being an issue, in this case entire tx should revert. I basically want to return an error ack if the error is unrecoverable (i.e. the callback itself is returning an error)

Reply:
Though it would be strange if you can try/catch a gas error right? Seems like a loophole that wouldn't exist 😄 

Trying to read through the docs to see what the behavior of gas panics in EVM is

Reply:
I think the gas issue is only relevant if we call the external contract with a specific amount of gas. With a normal call we would forward all the gas the tx has, which means that we would be out of gas in the catch (therefore never get to do anything anyway). If we had limited the call to some amount of gas, then we could in theory catch it here and handle it (if we had gas left).